### ISSUE TITLE
CI hanging up on Unit Tests

### ISSUE THREAD

**Issue by myieye** [sillsdev/languageforge-lexbox#325]
[Example run](https://github.com/sillsdev/languageforge-lexbox/actions/runs/6494496679/job/17637531194) timing out.
[Example](https://github.com/sillsdev/languageforge-lexbox/actions/runs/6493600421/job/17634873345) with slightly different output.

Last ["run without timeout"](https://github.com/sillsdev/languageforge-lexbox/actions/runs/6479803521/job/17594058543).

![image](https://github.com/sillsdev/languageforge-lexbox/assets/12587509/ecf14457-1faf-450e-a352-aead1a903622)

**Comment by myieye** [comment]
The [last run](https://github.com/sillsdev/languageforge-lexbox/actions/runs/6496090235) didn't get stuck, so either there's something weird going on in GitHub or we actually have a test that hangs.

![image](https://github.com/sillsdev/languageforge-lexbox/assets/12587509/efb24e06-bf59-4c5f-9a2e-bfa639a1d942)

**Comment by hahn-kev** [comment]
that's really weird. I'll enable the blame hang option on our tests to try to find the cause.

**Comment by myieye** [comment]
I looked into the hang dump:
Here's the juicy stuff from ` dotnet-dump analyze .\dotnet_2443_20231019T114120_hangdump.dmp -c dumpasync`:

```
STACK 3
<< Awaiting: 00007f98f0d922b0 00007fd906388f20 System.Runtime.CompilerServices.ValueTaskAwaiter<System.Boolean> >>
  00007f98f0d929c0 00007fd910745f38 (0) LexBoxApi.Services.RepoMigrationService+<WaitMigrationFinishedAsync>d__17
    00007f98f0d83160 00007fd9108fad78 (4) Testing.LexCore.Services.RepoMigrationTests+<CanNotSendReceiveWhenMigrating>d__20
```

**Comment by hahn-kev** [comment]
Fixed in b6639e6730560d771943e30e607c52841929673c

I also wrote some extra tests and removed any use of `CancellationToken.None` with a timer which means even if there's a deadlock in the future the test will fail with a timeout.
